homeassistant/helpers/entity.py

```diff
@@ -66,9 +65,12 @@ class Entity(object):
     # this class. These may be used to customize the behavior of the entity.
     entity_id = None  # type: str
 
-    # Owning hass instance. Will be set by EntityComponent
+    # Owning hass instance. Will be set by EntityPlatform
     hass = None  # type: Optional[HomeAssistant]
 
+    # Owning platform instance. Will be set by EntityPlatform
+    platform = None
+
```

Comment:
I think we should extend this entity with:
```
async def async_added_to_hass(self):
    pass

async def async_remove_from_hass(self):
    pass
```

Reply:
I rather have us check if it exists so that we don't need an extra yield? (we can do this in a later PR, will merge now)

Reply:
I think it make the object more understandable and help to give extra time for loop to handle parallel tasks. Like OS scheduler, more small task are better as long running tasks.